File: reference-code/src/main/java/createcommands/arguments/types/misc/SoundArguments.java
package createcommands.arguments.types.misc;

import dev.jorel.commandapi.CommandAPICommand;
import dev.jorel.commandapi.arguments.SoundArgument;
import org.bukkit.NamespacedKey;
import org.bukkit.Sound;

class SoundArguments {
    static {
        // #region soundArgumentsExample
        new CommandAPICommand("sound")
            .withArguments(new SoundArgument("sound"))
            .executesPlayer((player, args) -> {
                player.getWorld().playSound(player.getLocation(), (Sound) args.get("sound"), 100.0f, 1.0f);
            })
            .register();
        // #endregion soundArgumentsExample

        // #region soundArgumentsNamespacedKeyExample
        new CommandAPICommand("sound")
            .withArguments(new SoundArgument.NamespacedKey("sound"))
            .executesPlayer((player, args) -> {
                player.getWorld().playSound(player.getLocation(), ((NamespacedKey) args.get("sound")).asString(), 100.0f, 1.0f);
            })
            .register();
        // #endregion soundArgumentsNamespacedKeyExample
    }
}

File: reference-code/src/main/kotlin/createcommands/arguments/types/misc/SoundArguments.kt
package createcommands.arguments.types.misc

import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.arguments.SoundArgument
import dev.jorel.commandapi.executors.PlayerCommandExecutor
import dev.jorel.commandapi.kotlindsl.commandAPICommand
import dev.jorel.commandapi.kotlindsl.playerExecutor
import dev.jorel.commandapi.kotlindsl.soundArgument
import org.bukkit.NamespacedKey
import org.bukkit.Sound

fun soundArguments() {
    // #region soundArgumentsExample
    CommandAPICommand("sound")
        .withArguments(SoundArgument("sound"))
        .executesPlayer(PlayerCommandExecutor { player, args ->
            player.world.playSound(player.location, args["sound"] as Sound, 100.0f, 1.0f)
        })
        .register()
    // #endregion soundArgumentsExample

    // #region soundArgumentsNamespacedKeyExample
    CommandAPICommand("sound")
        .withArguments(SoundArgument.NamespacedKey("sound"))
        .executesPlayer(PlayerCommandExecutor { player, args ->
            player.world.playSound(player.location, (args["sound"] as NamespacedKey).asString(), 100.0f, 1.0f)
        })
        .register()
    // #endregion soundArgumentsNamespacedKeyExample
}

fun soundArgumentsDSL() {
    // #region soundArgumentsExampleDSL
    commandAPICommand("sound") {
        soundArgument("sound")
        playerExecutor { player, args ->
            player.world.playSound(player.location, args["sound"] as Sound, 100.0f, 1.0f)
        }
    }
    // #endregion soundArgumentsExampleDSL

    // #region soundArgumentsNamespacedKeyExampleDSL
    commandAPICommand("sound") {
        soundArgument("sound", true)
        playerExecutor { player, args ->
            player.world.playSound(player.location, (args["sound"] as NamespacedKey).asString(), 100.0f, 1.0f)
        }
    }
    // #endregion soundArgumentsNamespacedKeyExampleDSL
}